cloudferry_devlab/cloudferry_devlab/tests/functional_test.py

```diff
@@ -61,9 +62,15 @@ def setUp(self):
         self.migration_utils = utils.MigrationUtils(config)
         self.config_ini_path = config_ini['general']['configuration_ini_path']
         self.cloudferry_dir = config_ini['general']['cloudferry_dir']
-        self.filtering_utils = utils.FilteringUtils(
-            os.path.join(self.cloudferry_dir, get_option_from_config_ini(
-                'filter_path')))
+        tenant = get_nosetest_cmd_attribute_val('migrated_tenant')
+        if not tenant:
+            self.filtering_utils = utils.FilteringUtils(
+                os.path.join(self.cloudferry_dir,
+                             config.all_tenants_filter_filename))
```

Comment:
Split the update_filtering_file method and refactored its code to take care of generating yaml files per tenant and for all tenants.